packages/flame/lib/src/timer.dart

```diff
@@ -41,20 +46,29 @@ class Timer {
       if (_current >= limit) {
         if (!repeat) {
           _running = false;
-          onTick?.call();
+          _callTicker();
           return;
         }
         // This is used to cover the rare case of _current being more than
         // two times the value of limit, so that the onTick is called the
         // correct number of times
         while (_current >= limit) {
           _current -= limit;
-          onTick?.call();
+          _callTicker();
         }
       }
     }
   }
 
+  void _callTicker() {
+    if (tickCount != null && _currentTick >= tickCount!) {
```

Comment:
Shouldn't we stop directly after the last tick has been made, instead of in the beginning of the next tick?

Reply:
Good point, updated!

Reply:
If tick count is 0 for whatever reason we'd still tick now :)

Reply:
hum, yeah, but how should we handle this cause that really doesn't makes much sense right?

In the `TimerComponent` I  added the following assertion

```dart
assert(
  tickCount == null || tickCount > 0,
  'tickCount must be null or bigger than 0',
);
```

But I forgot to move it here when I moved the logic to the `Timer` class instead. Do you think moving the assertion here would be enough?

Reply:
That that sounds good!

Reply:
Moved the assertion 🥳